File: vigia-api/Vigia.API/Models/DTOs/Devices/DeviceDTO.cs
using Vigia.Models.Enums;

namespace Vigia.API.Models.DTOs.Devices;

public class DeviceDTO
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Nickname { get; set; }
    public DeviceRooms? Room { get; set; }
    public Guid? OwnerId { get; set; }
    public string MacAddress { get; set; } = string.Empty;
    public string? ThumbnailUrl { get; set; }
    public bool IsRunning { get; set; }
    public bool IsClipsEnabled { get; set; }
}

File: vigia-web/src/app/core/entities/DTOs/device.dto.ts
import { DeviceRooms } from '@core/enums';

export interface DeviceDto {
  id: string;
  name: string;
  nickname?: string | null;
  room?: DeviceRooms | string | null;
  ownerId?: string | null;
  macAddress: string;
  thumbnailUrl?: string | null;
  isRunning: boolean;
  isClipsEnabled: boolean;
}
